linkerd/core/src/main/scala/io/buoyant/linkerd/MetricsExport.scala

```diff
@@ -0,0 +1,24 @@
+package io.buoyant.linkerd
+
+import com.twitter.common.metrics.MetricProvider
+import com.twitter.finagle.stats.MetricsStatsReceiver
+import com.twitter.finagle.util.DefaultTimer
+import com.twitter.util.{Timer, Future, Time, Closable}
+import scala.collection.JavaConverters.mapAsScalaMapConverter
+
+class MetricsExport(
+  exporters: Seq[MetricsExporterConfig],
+  metrics: MetricProvider = MetricsStatsReceiver.defaultRegistry,
+  timer: Timer = DefaultTimer.twitter
+) extends Closable {
+
+  private[this] val tasks = exporters.map { exporter =>
+    timer.schedule(exporter.period) {
+      val _ = exporter.export(metrics.sample().asScala)
+    }
+  }
```

Comment:
That said, it definitely seems like a good idea to provide PeriodicMetricsExporter and PeriodicMetricsExporterConfig types to make it easy to write these.